### DISCUSSION TITLE
Overhead of method access

### DISCUSSION THREAD

**Discussion by juanjosegarciaripoll** [pybind/pybind11#5120] [Q&A]
Hi, I am in the process of migrating a rather large project from Python + Cython to PyBind11. However, I have found that the pybind11 code has a rather substantial overhead in class accessors. I have cooked a minimal example here https://github.com/juanjosegarciaripoll/pybind11_example When executed on a Windows AMD64 machine using the latest Miniforge and pybind 2.12, I get these timings
```
PyBind11 field 6.1339837000123225s, 6.13398e-07s per instance
PyBind11 method 7.244561500032432s, 7.24456e-07s per instance
PyBind11 unsafe method 5.384321800025646s, 5.38432e-07s per instance
Cython 1.1255456000217237s, 1.12555e-07s per instance
```
Am I doing anything wrong? Can I get closer to the Cython class example?

**Comment by wjakob** [comment]
Please take a look at https://nanobind.readthedocs.io/en/latest/. I started this project to reduce exactly those kinds of overheads.

**Reply by juanjosegarciaripoll** [reply]
Thanks. It is clear from this webpage why Pybind11 has a greater overhead than Cython. I will have a look and compare.
